pytest_minio_mock/plugin.py

```diff
@@ -322,11 +333,35 @@ def get_object(
         """
         self._health_check()
         the_object = self.buckets[bucket_name][object_name]
+        if (
```

Comment:
If a dictionary is always used as I suggested in my review for `put_object()` where a version_id is always used. then get_object can be simplified by doing the following:

    object_versions = self.buckets[bucket_name][object_name]
    
    # we try to get the version_id
    if  self.get_bucket_versioning(bucket_name).status != ENABLED
        version_id = list(the_object.keys())[-1]
    else:  # versionning is enabled
        if not version_id:
            version_id = list(the_object.keys())[-1]
            
    # we get the object using the version we got above
    try:
        the_object = object_versions[version_id]    
    except KeyError:
                raise S3Error(
                    message="The specified version does not exist",
                    resource=version_id,
                    request_id=None,
                    host_id=None,
                    response="mocked_response",
                    code=404,
                    bucket_name=bucket_name,
                    object_name=object_name,
                )